main.go

```diff
@@ -4,25 +4,45 @@ import (
 	"context"
 	"log"
 	"net/http"
+	"server/config"
 	"sync"
 	"time"
 
 	"server/internal/api"
 	"server/internal/middleware"
+
+	redis "github.com/dicedb/go-dice"
 )
 
 type HTTPServer struct {
 	httpServer *http.Server
+	diceClient *redis.Client
 }
 
-func NewHTTPServer(addr string, mux *http.ServeMux) *HTTPServer {
+func NewHTTPServer(addr string, mux *http.ServeMux, client *redis.Client) *HTTPServer {
```

Comment:
Let not use redis wording, we can change this to `*dice.Client`

Reply:
I have made the changes to replace all instances of 'Redis' with 'dice.Client' throughout the code and documentation. Everything is now updated accordingly.